pkg/monitoring/rules/recordingrules/operator.go

```diff
@@ -23,4 +23,12 @@ var operatorRecordingRules = []operatorrules.RecordingRule{
 		MetricType: operatormetrics.GaugeType,
 		Expr:       intstr.FromString(`sum(kube_pod_container_resource_requests{resource="cpu"} and on (pod) kube_pod_status_phase{phase="Running"} * on (pod) group_left kube_pod_labels{ label_kubevirt_io="virt-launcher"} > 0)`),
 	},
+	{
+		MetricsOpts: operatormetrics.MetricOpts{
+			Name: "cnv_abnormal",
+			Help: "Monitors resources for potential problems",
+		},
+		MetricType: operatormetrics.GaugeType,
+		Expr:       intstr.FromString("kubevirt_memory_delta_from_requested_bytes"),
```

Comment:
Can you elaborate on the reasoning behind the expression:

- Why are we using the delta of the requested bytes as a value for `cnv_abnormal` and how does someone seeing `cnv_abnormal` -> "Monitors resources for potential problems" will be able to understand what the value means?

- What happens when we want to add other metrics to this expression?

Reply:
additionally, since this is going to be a telemetry signal, please aggregate this to remove unneeded labels.

Reply:
@machadovilaca Regarding the first question. we add a label named `reason ` in the rule itself, see https://github.com/kubevirt/kubevirt/blob/main/pkg/monitoring/rules/recordingrules/operator.go#L31.
Regarding the second question- we will add the reason label also to other rules we want to add, then we add them to cnv_abnormal by using `or`.

@jan--f I don't see any label we can remove.. this rule as for now includes 8 records, see:

![Screenshot from 2024-04-10 14-06-36](https://github.com/kubevirt/hyperconverged-cluster-operator/assets/64130977/ca925717-0f22-4b8b-a941-e4a05c35f0b8) 
WDYT?

Reply:
Is the `namespace` label really useful here? Typically `container` and `node` are removed as well.

Even in the case these labels are useful for you, adding a recording rule with an explicitly label set to aggregate by adds context of what is supposed to be send and protects against later label additions getting send to telemetry inadvertently.

Reply:
@avlitman Please add the labels to the metric aggregation for visibility and as @jan--f mentioned, for making sure we only add these labels.

Reply:
Hi @sradco @machadovilaca @jan--f I removed the namespace label, and added `description` lable set to `memory_delta_from_requested_bytes`.